open-api/rest-catalog-open-api.yaml

```diff
@@ -1610,13 +1610,26 @@ components:
 
     PageToken:
       description:
-        An opaque token which allows clients to make use of pagination for a list API (e.g. ListTables). 
-        Clients will initiate the first paginated request by sending an empty `pageToken` e.g. `GET /tables?pageToken` or `GET /tables?pageToken=`
-        signaling to the service that the response should be paginated.
-
-        Servers that support pagination will recognize `pageToken` and return a `next-page-token` in response if there are more results available.
-        After the initial request, it is expected that the value of `next-page-token` from the last response is used in the subsequent request.
-        Servers that do not support pagination will ignore `next-page-token` and return all results.
+        An opaque token that allows clients to make use of pagination for list APIs 
+        (e.g. ListTables). Clients may initiate the first paginated request by sending an empty 
+        query parameter `pageToken` to the server.
+
+        Servers that support pagination should identify the `pageToken` parameter and return a 
+        `next-page-token` in the response if there are more results available.  After the initial 
+        request, the value of `next-page-token` from each response must be used as the `pageToken` 
+        parameter value for the next request.  The server must return `null` value for the
+        `next-page-token` in the last response.
+        
+        Servers that support pagination must return all results in a single response with the value
```

Comment:
How does the service distinguish this case from the initial request when `pageToken` is empty, but we still return the number of items determined by `pageSize`?

Reply:
If no token is provided, I don't think pageSize makes sense because there's no guaranteed ordering.  I think it's correct to say that if pageToken is not present, all results are returned.